frappe/public/js/frappe/model/create_new.js

```diff
@@ -292,7 +292,7 @@ $.extend(frappe.model, {
 			opts.source_name = opts.frm.doc.name;
 
 		// Allow opening a mapped doc without a source document name
-		} else if (!opts.frm) {
+		} else if (!opts.frm && !opts.source_name) {
 			opts.source_name = null;
 		}
```

Comment:
This block of code does not make sense... 
According to the code we are setting `source_name` as `null` if source_name is `null/undefined`?